apps/api/src/workspace-role/workspace-role.e2e.spec.ts

```diff
@@ -366,6 +366,85 @@ describe('Workspace Role Controller Tests', () => {
 
       expect(response.statusCode).toBe(401)
     })
+
+    it('should be able to create workspace role with environment only access for projects', async () => {
+      const devEnvironment = await prisma.environment.create({
+        data: {
+          name: 'development',
+          slug: 'development',
+          projectId: projects[0].id
+        }
+      })
+      const response = await app.inject({
+        method: 'POST',
+        url: `/workspace-role/${workspaceAlice.slug}`,
+        payload: {
+          name: 'Test Role 2',
+          description: 'Test Role Description',
+          colorCode: '#0000FF',
+          authorities: [Authority.READ_ENVIRONMENT, Authority.READ_VARIABLE],
+          projectEnvironments: [
+            {
+              projectSlug: projects[0].slug,
+              environmentSlugs: ['development']
+            }
+          ]
+        },
+        headers: {
+          'x-e2e-user-email': alice.email
+        }
+      })
+      expect(response.statusCode).toBe(201)
+      expect(response.json()).toEqual(
+        expect.objectContaining({
+          id: expect.any(String),
+          name: 'Test Role 2',
+          description: 'Test Role Description',
+          colorCode: '#0000FF',
+          authorities: [Authority.READ_ENVIRONMENT, Authority.READ_VARIABLE],
+          workspaceId: workspaceAlice.id,
+          projects: [
+            {
+              project: {
+                id: projects[0].id,
+                name: projects[0].name,
+                slug: projects[0].slug
+              },
+              environments: [
+                {
+                  id: devEnvironment.id,
+                  name: 'development',
+                  slug: 'development'
+                }
+              ]
+            }
+          ]
+        })
+      )
+    })
+
```

Comment:
Another case should be - should not be able to create workspace role where user doesn't have access to environment.